### PR TITLE
[doc] Document how to contribute

### PR DESCRIPTION
**Pull request by JingsongLi**
Refer to https://flink.apache.org/how-to-contribute/overview/

### CODE REVIEW

File: docs/content/project/contributing.md [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+# Contributing
+
+Apache Paimon (incubating) is developed by an open and friendly community. Everybody is cordially welcome to join
```
Comment (SteNicholas) [inline docs/content/project/contributing.md:29]:
```suggestion
Apache Paimon (incubating) is developed by an open and friendly community. Everybody is cordially welcomed to join
```

File: docs/content/project/contributing.md [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+the community and contribute to Apache Paimon. There are several ways to interact with the community and to contribute
```
Comment (SteNicholas) [inline docs/content/project/contributing.md:30]:
```suggestion
the community and contribute to Apache Paimon. There are several ways to interact with the community and contribute
```

File: docs/content/project/contributing.md [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+to Paimon including asking questions, filing bug reports, proposing new features, joining discussions on the mailing
```
Comment (SteNicholas) [inline docs/content/project/contributing.md:31]:
```suggestion
to Paimon including raising issues, filing bug reports, proposing new features, joining discussions on the mailing
```

File: docs/content/project/contributing.md [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+lists, contributing code or documentation, improving the website, or testing release candidates.
```
Comment (SteNicholas) [inline docs/content/project/contributing.md:32]:
```suggestion
lists, contributing code or documentation, improving website, testing release candidates and writing corresponding blog etc.
```

File: docs/content/project/contributing.md [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+## Code Review Guide
+
+Every review needs to check the following six aspects. **We encourage to check these aspects in order, to avoid
+spending time on detailed code quality reviews when formal requirements are not met or there is no consensus in
+the community to accept the change.**
+
+#### 1. Is the Contribution Well-Described?
+
+Check whether the contribution is sufficiently well-described to support a good review. Trivial changes and fixes
+do not need a long description. If the implementation is exactly according to a prior discussion on Issue or the
```
Comment (SteNicholas) [inline docs/content/project/contributing.md:199]:
```suggestion
do not need a long description. If the implementation is exactly according to a prior discussion on issue or the
```

File: docs/content/project/contributing.md [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+
+Some changes require attention and approval from specific committers.
+
+If the pull request needs specific attention, one of the tagged committers/contributors should give the final approval.
+
+#### 3. Is the Overall Code Quality Good, Meeting Standard we Want to Maintain in Paimon?
+
+This is the detailed code review of the actual changes, covering:
+
+- Are the changes doing what is described in the Issue or design document?
```
Comment (SteNicholas) [inline docs/content/project/contributing.md:215]:
```suggestion
- Are the changes doing what is described in the issue or design document?
```

**Review by SteNicholas — CHANGES_REQUESTED** [review]
@JingsongLi, I left some comments for contribution document. PTAL.

File: docs/content/project/contributing.md [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+If the implementation is different from the agreed approach in the consensus discussion, a detailed description of
+the implementation is required for any further review of the contribution.
+
+#### 2. Does the Contribution Need Attention from some Specific Committers?
+
+Some changes require attention and approval from specific committers.
+
+If the pull request needs specific attention, one of the tagged committers/contributors should give the final approval.
+
+#### 3. Is the Overall Code Quality Good, Meeting Standard we Want to Maintain in Paimon?
```
Comment (Myasuka) [inline docs/content/project/contributing.md:211]: I think the description of this part is a bit ambiguous. How can we define good code quality? I think we can add reference to [Flink's code guide](https://flink.apache.org/how-to-contribute/code-style-and-quality-preamble/) before Paimon already supports one.

File: docs/content/project/contributing.md [hunk 8]
```diff
@@ -0,0 +1,253 @@
...
+        <p><b>To request an issue, please note that it is not just a "please assign it to me", you need to explain your understanding of the issue, and your design, and if possible, you need to provide your POC code.</b></p>
+      </div>
+    </div>
+  </div>
+  <div class="column">
+    <div class="panel panel-default">
+      <div class="panel-body">
+        <h2><span class="number">2</span><a href="#implement">Implement</a></h2>
+        <p>Create the Pull Request and the approach agreed upon in the issue.</p>
+        <p><b>1.Only create the PR if you are assigned to the issue. 2.Please associate an issue (if any), e.g. fix #123. 3.Please enable the actions of your own clone project, show a screenshot of the test passing.</b></p>
```
Comment (Myasuka) [inline docs/content/project/contributing.md:167]: Why we must need to `show a screenshot of the test passing`?

**Review by Myasuka — COMMENTED** [review]
Since more and more developers involved, I think a guide of contributing is very important. Please take a look at my comments.

File: docs/content/project/contributing.md (hunk 1 above)
Comment (JingsongLi) [reply docs/content/project/contributing.md:29]: welcome is an adjective, here can be `welcome`

File: docs/content/project/contributing.md (hunk 3 above)
Comment (JingsongLi) [reply docs/content/project/contributing.md:31]: `raising issues` contains `filing bug reports`?

File: docs/content/project/contributing.md (hunk 7 above)
Comment (JingsongLi) [reply docs/content/project/contributing.md:211]: good point, add `Code guidelines can be found in the [Flink Code Style and Quality Guide](https://flink.apache.org/how-to-contribute/code-style-and-quality-preamble/).`

File: docs/content/project/contributing.md (hunk 8 above)
Comment (JingsongLi) [reply docs/content/project/contributing.md:167]: For non-apache-committers, only testing run in their own github repo. It is good to show the test passing.

Comment (JingsongLi) [reply docs/content/project/contributing.md:167]: OK it is ugly, reviewer can take a look to corresponding link to see test passing.

**Review by Myasuka — APPROVED** [review]
LGTM, thanks for the update.

**Review by SteNicholas — APPROVED** [review]
LGTM.

**Pull request merged**